### PR TITLE
Release pipeline

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

Release pipeline for `graql`, which performs:
1. deployment of a Github draft release
2. Deployment of artifacts to Maven

### CODE REVIEW

File: deployment.properties [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+# TODO: Is having a custom deployment.properties like this a good idea?
+
+# TODO: replace back to 'graknlabs'
+repo.github.organisation=lolski
+repo.github.repository=graql
```
Comment (lolski) [inline deployment.properties]: need to be supplied via the rule instead of `deployment.properties`

File: deployment.properties [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+repo.maven.snapshot=https://repo.grakn.ai/repository/test-maven/
+repo.maven.release=https://repo.grakn.ai/repository/test-maven/
```
Comment (lolski) [inline deployment.properties]: need to be changed to maven.central

File: deployment.properties [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (lolski) [inline deployment.properties]: shouldn't have to have this file here

File: .circleci/config.yml [hunk 4]
```diff
@@ -71,22 +71,85 @@ jobs:
...
+      - checkout
+      - run: bazel run @graknlabs_build_tools//ci:release-approval
+
+  deploy-github:
+    machine: true
+    working_directory: ~/grakn
+    steps:
+      - install-bazel-linux-rbe
+      - checkout
+      - run: bazel run //java:deploy-github -- ${GRABL_CREDENTIAL}
```
Comment (lolski) [inline .circleci/config.yml]: credential should be passed in as environment variables

File: java/BUILD [hunk 5]
```diff
@@ -20,6 +20,8 @@ package(default_visibility = ["//visibility:public"])
 
 load("@graknlabs_build_tools//checkstyle:rules.bzl", "checkstyle_test")
 load("@graknlabs_build_tools//distribution/maven:rules.bzl", "assemble_maven", "deploy_maven")
+load("@graknlabs_bazel_distribution//common:rules.bzl", "assemble_zip", "java_deps")
```
Comment (lolski) [inline java/BUILD]: no need to import `common:rules.bzl`

File: WORKSPACE [hunk 6]
```diff
@@ -83,6 +83,29 @@ load("@rules_antlr//antlr:deps.bzl", "antlr_dependencies")
...
+# TODO: rename the macro we load here to deploy_github_dependencies
+load("@graknlabs_bazel_distribution//github:dependencies.bzl", "github_dependencies_for_deployment")
+github_dependencies_for_deployment()
+
+load("@graknlabs_build_tools//bazel:dependencies.bzl", "bazel_rules_docker")
+bazel_rules_docker()
+
+load("@io_bazel_rules_docker//repositories:repositories.bzl",
+bazel_rules_docker_repositories = "repositories")
+bazel_rules_docker_repositories()
```
Comment (haikalpribadi) [inline WORKSPACE]: why do we need to load all these docker dependencies, @lolski?
- `bazel_rules_docker()`
- `bazel_rules_docker_repositories()`
- `container_pull()`

Is there anything in `@graknlabs_graql` that requires Docker?

Comment (lolski) [reply WORKSPACE]: good point, I thought they were transitive dependencies but turns out they're not.

Comment (lolski) [reply WORKSPACE]: fixed in 8d1dfa2a9055b3f9ed07e7ffe7c94c8421711400

**Pull request merged**